planner/core/logical_plan_builder.go

```diff
@@ -3631,6 +3631,11 @@ func (b *PlanBuilder) pushTableHints(hints []*ast.TableOptimizerHint, currentLev
 			limitHints.preferLimitToCop = true
 		case HintMerge:
 			MergeHints.preferMerge = true
+			if hint.Tables != nil {
+				b.ctx.GetSessionVars().StmtCtx.AppendWarning(ErrInternal.GenWithStack("The MERGE hint is not used correctly, maybe it inputs a table name."))
+				MergeHints.preferMerge = false
```

Comment:
You can put `MergeHints.preferMerge = true` after L3638, and remove this line.

Reply:
ok